server/index-node/src/explorer.rs

```diff
@@ -0,0 +1,222 @@
+            synced: vi.synced,
+            failed: vi.failed,
+            description: vi.description.as_ref().map(|s| s.as_str()),
+            repository: vi.repository.as_ref().map(|s| s.as_str()),
+            schema: vi.schema.document.to_string(),
+            network: vi.network.as_ref().map(|s| s.as_str())
+        };
+        Ok(as_http_response(&value))
+    }
+
+    fn handle_subgraph_repo(&self, version: &str) -> Result<Response<Body>, GraphQLServerError> {
+        let vi = self.version_info(version)?;
+
+        let value = object! {
+            createdAt: vi.created_at.as_str(),
+            deploymentId: vi.deployment_id.as_str(),
+            repository: vi.repository.as_ref().map(|s| s.as_str())
+        };
+        Ok(as_http_response(&value))
+    }
+
+    fn handle_entity_count(&self, deployment: &str) -> Result<Response<Body>, GraphQLServerError> {
+        if let Some(value) = self.entity_counts.get(deployment) {
+            return Ok(as_http_response(value.as_ref()));
+        }
+
+        let infos = self
+            .store
+            .status(status::Filter::Deployments(vec![deployment.to_string()]))?;
+        let info = match infos.first() {
+            Some(info) => info,
+            None => {
+                return handle_not_found();
+            }
+        };
+
+        let value = object! {
+            entityCount: info.entity_count
+        };
+        let resp = as_http_response(&value);
+        self.entity_counts
+            .set(deployment.to_string(), Arc::new(value));
+        Ok(resp)
+    }
+
+    fn version_info(&self, version: &str) -> Result<Arc<VersionInfo>, GraphQLServerError> {
+        match self.version_infos.get(version) {
+            Some(vi) => Ok(vi),
+            None => {
+                let vi = Arc::new(self.store.version_info(version)?);
+                self.version_infos.set(version.to_string(), vi.clone());
+                Ok(vi)
+            }
+        }
+    }
+}
+
+fn handle_not_found() -> Result<Response<Body>, GraphQLServerError> {
+    Ok(Response::builder()
+        .status(StatusCode::NOT_FOUND)
+        .body(Body::from("Not found"))
+        .unwrap())
+}
+
+fn as_http_response(value: &q::Value) -> http::Response<Body> {
+    let status_code = http::StatusCode::OK;
+    let json = serde_json::to_string(&SerializableValue(value))
+        .expect("Failed to serialize response to JSON");
+    http::Response::builder()
+        .status(status_code)
+        .header("Access-Control-Allow-Origin", "*")
+        .header("Access-Control-Allow-Headers", "Content-Type, User-Agent")
+        .header("Access-Control-Allow-Methods", "GET, OPTIONS, POST")
+        .header("Content-Type", "application/json")
+        .body(Body::from(json))
+        .unwrap()
+}
+
+/// Caching of values for a specified amount of time
+#[derive(Debug)]
+struct CacheEntry<T> {
+    value: Arc<T>,
+    expires: Instant,
+}
+
+#[derive(Debug)]
+struct Cache<T> {
```

Comment:
Worth a comment that this does not ever remove entries, but starts returning `None` after the TTL has elapsed.

Reply:
Good point, added comments